components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -551,4 +584,55 @@ private String getChannelTypeFromResource(Resource resource) {
                     .map(Attribute::getValue).orElse(DEFAULT_HANDLER_NAME);
         }
     }
+
+    /**
+     * Method to check whether the sender is allowed to delete.
+     *
+     * @param senderName    Name of the sender.
+     * @return  whether the sender is allowed to delete.
+     * @throws NotificationSenderManagementException If an error occurred while checking the sender.
+     */
+    private boolean checkAllowDelete(String senderName) throws NotificationSenderManagementException {
+
+        if (SENDERS.get(senderName) != null) {
```

Comment:
What will happen  if (SENDERS.get(senderName) == null), we need to return false right?

Reply:
WE introduce a map `SENDERS` to have the senderName (we use in the API) and the relevant authenticator name. Added a map that we can use this check method in deleting or disabling the authenticators that we may introduce in future. If the senderName is something else, it will be handled in the actual method, so we do not need to do the validations (whether there are connected app) for them.